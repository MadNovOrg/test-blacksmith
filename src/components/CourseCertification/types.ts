export type ModuleObject = {
  name: string
  completed: boolean
}
