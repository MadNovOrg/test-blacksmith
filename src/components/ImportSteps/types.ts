import { ComponentProps } from 'react'

import { StepsNavigation } from '../StepsNavigation'

export enum ImportStepsEnum {
  CHOOSE = 'choose',
  CONFIGURE = 'configure',
  PREVIEW = 'preview',
  IMPORTING = 'importing',
  RESULTS = 'results',
}
export type Step = {
  label: ComponentProps<typeof StepsNavigation>['steps'][0]['label']
  key: ImportStepsEnum
}
