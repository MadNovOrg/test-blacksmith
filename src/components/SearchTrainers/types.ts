import type { InviteStatus, CourseSchedule } from '@app/types'

export type SearchTrainersSchedule = Partial<
  Pick<CourseSchedule, 'start' | 'end'>
>

export type SearchTrainerBookings = {
  profile_id: string
  status: InviteStatus
  course: {
    schedule: Array<{
      start: string
      end: string
    }>
  }
}
