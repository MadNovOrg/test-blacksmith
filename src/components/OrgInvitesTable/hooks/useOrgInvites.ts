import { useCallback, useMemo } from 'react'
import { gql, useMutation, useQuery } from 'urql'

import {
  DeleteOrgInviteMutation,
  DeleteOrgInviteMutationVariables,
  GetOrgInvitesQuery,
  GetOrgInvitesQueryVariables,
  RecreateOrgInviteMutation,
  RecreateOrgInviteMutationVariables,
} from '@app/generated/graphql'
import { PROFILE } from '@app/queries/fragments'

export const GET_ORG_INVITES = gql`
  ${PROFILE}
  query GetOrgInvites(
    $orgId: uuid!
    $limit: Int = 20
    $offset: Int = 0
    $where: organization_invites_bool_exp = {}
  ) {
    orgInvites: organization_invites(
      where: { _and: [{ orgId: { _eq: $orgId } }, $where] }
      limit: $limit
      offset: $offset
      order_by: { createdAt: desc }
    ) {
      id
      createdAt
      updatedAt
      email
      status
      isAdmin
      profile {
        ...Profile
      }
      organization {
        ...Organization
      }
    }
    total: organization_invites_aggregate(
      where: { _and: [{ orgId: { _eq: $orgId } }, $where] }
    ) {
      aggregate {
        count
      }
    }
  }
`
export const DELETE_ORG_INVITE = gql`
  mutation DeleteOrgInvite($inviteId: uuid!) {
    delete_organization_invites_by_pk(id: $inviteId) {
      id
    }
  }
`
export const RESEND_ORG_INVITE = gql`
  mutation RecreateOrgInvite($inviteId: uuid!) {
    update_organization_invites_by_pk(
      pk_columns: { id: $inviteId }
      _set: { updatedAt: "now()" }
    ) {
      id
    }
  }
`

type Props = {
  limit?: number
  offset?: number
  where?: object
}

export const useOrgInvites = (
  orgId: string,
  { limit, offset, where }: Props,
) => {
  const [{ data, error, fetching }, mutate] = useQuery<
    GetOrgInvitesQuery,
    GetOrgInvitesQueryVariables
  >({
    query: GET_ORG_INVITES,
    variables: { orgId, limit, offset, where },
  })

  const [, resendOrgInvite] = useMutation<
    RecreateOrgInviteMutation,
    RecreateOrgInviteMutationVariables
  >(RESEND_ORG_INVITE)

  const [, deleteOrgInvite] = useMutation<
    DeleteOrgInviteMutation,
    DeleteOrgInviteMutationVariables
  >(DELETE_ORG_INVITE)

  const resend = useCallback(
    async (inviteId: string) => {
      await resendOrgInvite({
        inviteId: inviteId,
      })
      mutate()
    },
    [mutate, resendOrgInvite],
  )

  const cancel = useCallback(
    async (inviteId: string) => {
      await deleteOrgInvite({
        inviteId: inviteId,
      })
      mutate()
    },
    [deleteOrgInvite, mutate],
  )

  return useMemo(
    () => ({
      invites: data?.orgInvites ?? [],
      totalCount: data?.total.aggregate?.count ?? 0,
      resend,
      cancel,
      error,
      loading: fetching,
      mutate,
    }),
    [
      data?.orgInvites,
      data?.total.aggregate?.count,
      resend,
      cancel,
      error,
      fetching,
      mutate,
    ],
  )
}
