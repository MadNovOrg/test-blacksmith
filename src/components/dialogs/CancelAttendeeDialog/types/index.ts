export type FormInput = {
  cancellationFeePercent: number
  cancellationReason: string
}
