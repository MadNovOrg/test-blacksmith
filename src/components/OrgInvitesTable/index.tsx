import {
  Box,
  Button,
  Chip,
  CircularProgress,
  Paper,
  Stack,
  Table,
  TableBody,
  TableCell,
  TableContainer,
  TablePagination,
  TableRow,
} from '@mui/material'
import MuiTableHead from '@mui/material/TableHead'
import { groupBy } from 'lodash-es'
import React, { ChangeEvent, useCallback, useMemo, useState } from 'react'
import { useTranslation } from 'react-i18next'

import { useOrgInvites } from '@app/components/OrgInvitesTable/hooks/useOrgInvites'
import { TableNoRows } from '@app/components/Table/TableNoRows'
import { Course_Invite_Status_Enum } from '@app/generated/graphql'
import {
  DEFAULT_PAGINATION_LIMIT,
  DEFAULT_PAGINATION_ROW_OPTIONS,
} from '@app/util'

type OrgInvitesTableParams = {
  orgId: string
}

export const OrgInvitesTable: React.FC<
  React.PropsWithChildren<OrgInvitesTableParams>
> = ({ orgId }) => {
  const { t } = useTranslation()

  const [currentPage, setCurrentPage] = useState(0)
  const [perPage, setPerPage] = useState(DEFAULT_PAGINATION_LIMIT)

  const { invites, loading, totalCount, resend, cancel } = useOrgInvites(
    orgId,
    {
      limit: perPage,
      offset: perPage * currentPage,
    },
  )

  const invitesByStatus = useMemo(() => {
    if (!invites) return []
    const byStatus = groupBy(
      invites,
      inv => inv.status === Course_Invite_Status_Enum.Pending,
    )
    return [...(byStatus['true'] ?? []), ...(byStatus['false'] ?? [])]
  }, [invites])

  const handleRowsPerPageChange = useCallback(
    (event: ChangeEvent<HTMLTextAreaElement | HTMLInputElement>) => {
      setPerPage(parseInt(event.target.value, 10))
      setCurrentPage(0)
    },
    [],
  )

  return (
    <>
      <TableContainer component={Paper} elevation={0}>
        <Table>
          <MuiTableHead>
            <TableRow>
              <TableCell>{t('common.email')}</TableCell>
              <TableCell>
                {t('pages.org-details.tabs.users.invited-on')}
              </TableCell>
              <TableCell>
                {t('pages.org-details.tabs.users.permissions')}
              </TableCell>
              <TableCell>{t('common.status')}</TableCell>
              <TableCell></TableCell>
            </TableRow>
          </MuiTableHead>
          <TableBody data-testid="org-users-table">
            {loading ? (
              <TableRow
                sx={{
                  '&&.MuiTableRow-root': {
                    backgroundColor: 'grey.300',
                  },
                  '&& .MuiTableCell-root': {
                    px: 2,
                    py: 1,
                    color: 'grey.700',
                    fontWeight: '600',
                  },
                }}
              >
                <TableCell colSpan={4}>
                  <Stack direction="row" alignItems="center">
                    <CircularProgress size={20} />
                  </Stack>
                </TableCell>
              </TableRow>
            ) : null}

            <TableNoRows
              noRecords={!loading && !invites?.length}
              itemsName={t('invites').toLowerCase()}
              colSpan={4}
            />

            {invitesByStatus.map(invite => {
              return (
                <TableRow key={invite.id}>
                  <TableCell>{invite.email}</TableCell>
                  <TableCell>
                    {t('dates.default', { date: invite.createdAt })}
                  </TableCell>
                  <TableCell>
                    <Chip
                      label={
                        invite.isAdmin
                          ? t('pages.org-details.tabs.users.organization-admin')
                          : t('pages.org-details.tabs.users.no-permissions')
                      }
                      color={invite.isAdmin ? 'success' : 'gray'}
                      size="small"
                    />
                  </TableCell>
                  <TableCell>
                    <Chip
                      label={t(
                        `pages.org-details.tabs.users.invite-status.${invite.status.toLowerCase()}`,
                      )}
                      color={
                        invite.status === Course_Invite_Status_Enum.Accepted
                          ? 'success'
                          : invite.status === Course_Invite_Status_Enum.Declined
                          ? 'error'
                          : 'secondary'
                      }
                      size="small"
                    />
                  </TableCell>
                  <TableCell>
                    {invite.status === Course_Invite_Status_Enum.Pending ? (
                      <Box display="flex" justifyContent="flex-end">
                        <Button
                          data-testid="resend-invite-button"
                          size="large"
                          variant="text"
                          color="primary"
                          onClick={() => resend(invite.id)}
                        >
                          {t('common.resend')}
                        </Button>
                        <Button
                          data-testid="cancel-invite-button"
                          size="large"
                          variant="text"
                          color="primary"
                          onClick={() => cancel(invite.id)}
                        >
                          {t('common.cancel-invite')}
                        </Button>
                      </Box>
                    ) : null}
                  </TableCell>
                </TableRow>
              )
            })}
          </TableBody>
        </Table>
      </TableContainer>
      {totalCount ? (
        <TablePagination
          component="div"
          count={totalCount}
          page={currentPage}
          onPageChange={(_, page) => setCurrentPage(page)}
          onRowsPerPageChange={handleRowsPerPageChange}
          rowsPerPage={perPage}
          rowsPerPageOptions={DEFAULT_PAGINATION_ROW_OPTIONS}
        />
      ) : null}
    </>
  )
}
