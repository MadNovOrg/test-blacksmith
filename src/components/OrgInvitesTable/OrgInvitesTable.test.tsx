import React from 'react'
import { Route, Routes } from 'react-router-dom'

import { OrgInvitesTable } from '@app/components/OrgInvitesTable/index'
import { useOrgInvites } from '@app/hooks/useOrgInvites'
import { InviteStatus } from '@app/types'
import { LoadingStatus } from '@app/util'

import { render } from '@test/index'

jest.mock('@app/hooks/useOrgInvites')

const useOrgInvitesMock = jest.mocked(useOrgInvites)

describe('OrgInvitesTable', () => {
  useOrgInvitesMock.mockReturnValue({
    mutate: () => Promise.resolve(undefined),
    status: LoadingStatus.SUCCESS,
    loading: false,
    totalCount: 2,
    error: undefined,
    cancel: jest.fn(),
    resend: jest.fn(),
    invites: [
      {
        id: '1',
        createdAt: new Date('2022-04-10T10:00:00').toISOString(),
        updatedAt: new Date('2022-04-10T10:00:00').toISOString(),
        email: 'email1@org.com',
        status: InviteStatus.PENDING,
        organization: {
          id: 'org-1',
          name: 'org-1',
        },
        isAdmin: false,
      },
      {
        id: '2',
        createdAt: new Date('2022-04-10T10:00:00').toISOString(),
        updatedAt: new Date('2022-04-10T10:00:00').toISOString(),
        email: 'email2@org.com',
        status: InviteStatus.DECLINED,
        organization: {
          id: 'org-1',
          name: 'org-1',
        },
        isAdmin: true,
      },
    ],
  })

  it('matches snapshot', async () => {
    const view = render(
      <Routes>
        <Route path="/" element={<OrgInvitesTable orgId="1" />} />
      </Routes>,
      {},
      { initialEntries: ['/'] }
    )

    expect(view).toMatchSnapshot()
  })
})
