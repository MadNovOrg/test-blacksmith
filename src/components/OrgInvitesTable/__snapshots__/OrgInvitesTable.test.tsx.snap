// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrgInvitesTable matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiTableContainer-root css-zbd12r-MuiPaper-root-MuiTableContainer-root"
      >
        <table
          class="MuiTable-root css-16un4wc-MuiTable-root"
        >
          <thead
            class="MuiTableHead-root css-kv444d-MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head css-555rmi-MuiTableRow-root"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-s9dfu1-MuiTableCell-root"
                scope="col"
              >
                Email
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-s9dfu1-MuiTableCell-root"
                scope="col"
              >
                Invited on
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-s9dfu1-MuiTableCell-root"
                scope="col"
              >
                Permissions
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-s9dfu1-MuiTableCell-root"
                scope="col"
              >
                Status
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-s9dfu1-MuiTableCell-root"
                scope="col"
              />
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root css-6wwm3k-MuiTableBody-root"
          >
            <tr
              class="MuiTableRow-root css-555rmi-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-11mkwu5-MuiTableCell-root"
              >
                email1@org.com
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-11mkwu5-MuiTableCell-root"
              >
                10 April 2022
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-11mkwu5-MuiTableCell-root"
              >
                <div
                  class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorSecondary MuiChip-filledSecondary css-447n4u-MuiChip-root"
                >
                  <span
                    class="MuiChip-label MuiChip-labelSmall css-t3ycia-MuiChip-label"
                  >
                    No permissions
                  </span>
                </div>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-11mkwu5-MuiTableCell-root"
              >
                <div
                  class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorSecondary MuiChip-filledSecondary css-447n4u-MuiChip-root"
                >
                  <span
                    class="MuiChip-label MuiChip-labelSmall css-t3ycia-MuiChip-label"
                  >
                    Pending
                  </span>
                </div>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-11mkwu5-MuiTableCell-root"
              >
                <div
                  class="MuiBox-root css-1bvc4cc"
                >
                  <button
                    class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeLarge MuiButton-textSizeLarge MuiButtonBase-root  css-zcpmcl-MuiButtonBase-root-MuiButton-root"
                    data-testid="resend-invite-button"
                    tabindex="0"
                    type="button"
                  >
                    Resend
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                  <button
                    class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeLarge MuiButton-textSizeLarge MuiButtonBase-root  css-zcpmcl-MuiButtonBase-root-MuiButton-root"
                    data-testid="cancel-invite-button"
                    tabindex="0"
                    type="button"
                  >
                    Cancel invite
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-555rmi-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-11mkwu5-MuiTableCell-root"
              >
                email2@org.com
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-11mkwu5-MuiTableCell-root"
              >
                10 April 2022
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-11mkwu5-MuiTableCell-root"
              >
                <div
                  class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorSuccess MuiChip-filledSuccess css-10gd6cu-MuiChip-root"
                >
                  <span
                    class="MuiChip-label MuiChip-labelSmall css-t3ycia-MuiChip-label"
                  >
                    Organisation admin
                  </span>
                </div>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-11mkwu5-MuiTableCell-root"
              >
                <div
                  class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorError MuiChip-filledError css-o7yv1p-MuiChip-root"
                >
                  <span
                    class="MuiChip-label MuiChip-labelSmall css-t3ycia-MuiChip-label"
                  >
                    Declined
                  </span>
                </div>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-11mkwu5-MuiTableCell-root"
              />
            </tr>
          </tbody>
        </table>
        <div
          class="MuiTablePagination-root css-jtlhu6-MuiTablePagination-root"
        >
          <div
            class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar css-17xoeoc-MuiToolbar-root-MuiTablePagination-toolbar"
          >
            <div
              class="MuiTablePagination-spacer css-1psng7p-MuiTablePagination-spacer"
            />
            <p
              class="MuiTablePagination-selectLabel css-1fki1wl-MuiTablePagination-selectLabel"
              id="mui-2"
            >
              Rows per page:
            </p>
            <div
              class="MuiInputBase-root MuiInputBase-colorPrimary css-i13sg4-MuiInputBase-root-MuiTablePagination-select"
              variant="standard"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="mui-2 mui-1"
                class="MuiTablePagination-select MuiSelect-select MuiSelect-standard MuiInputBase-input css-194a1fa-MuiSelect-select-MuiInputBase-input"
                id="mui-1"
                role="button"
                tabindex="0"
              >
                12
              </div>
              <input
                aria-hidden="true"
                class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                tabindex="-1"
                value="12"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTablePagination-selectIcon MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
            <p
              class="MuiTablePagination-displayedRows css-lwa6hs-MuiTablePagination-displayedRows"
            >
              1â€“2 of 2
            </p>
            <div
              class="MuiTablePagination-actions"
            >
              <button
                aria-label="Go to previous page"
                class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-1hk5nzv-MuiButtonBase-root-MuiIconButton-root"
                disabled=""
                tabindex="-1"
                title="Go to previous page"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="KeyboardArrowLeftIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                  />
                </svg>
              </button>
              <button
                aria-label="Go to next page"
                class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-1hk5nzv-MuiButtonBase-root-MuiIconButton-root"
                disabled=""
                tabindex="-1"
                title="Go to next page"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="KeyboardArrowRightIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiTableContainer-root css-zbd12r-MuiPaper-root-MuiTableContainer-root"
    >
      <table
        class="MuiTable-root css-16un4wc-MuiTable-root"
      >
        <thead
          class="MuiTableHead-root css-kv444d-MuiTableHead-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head css-555rmi-MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-s9dfu1-MuiTableCell-root"
              scope="col"
            >
              Email
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-s9dfu1-MuiTableCell-root"
              scope="col"
            >
              Invited on
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-s9dfu1-MuiTableCell-root"
              scope="col"
            >
              Permissions
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-s9dfu1-MuiTableCell-root"
              scope="col"
            >
              Status
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-s9dfu1-MuiTableCell-root"
              scope="col"
            />
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root css-6wwm3k-MuiTableBody-root"
        >
          <tr
            class="MuiTableRow-root css-555rmi-MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-11mkwu5-MuiTableCell-root"
            >
              email1@org.com
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-11mkwu5-MuiTableCell-root"
            >
              10 April 2022
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-11mkwu5-MuiTableCell-root"
            >
              <div
                class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorSecondary MuiChip-filledSecondary css-447n4u-MuiChip-root"
              >
                <span
                  class="MuiChip-label MuiChip-labelSmall css-t3ycia-MuiChip-label"
                >
                  No permissions
                </span>
              </div>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-11mkwu5-MuiTableCell-root"
            >
              <div
                class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorSecondary MuiChip-filledSecondary css-447n4u-MuiChip-root"
              >
                <span
                  class="MuiChip-label MuiChip-labelSmall css-t3ycia-MuiChip-label"
                >
                  Pending
                </span>
              </div>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-11mkwu5-MuiTableCell-root"
            >
              <div
                class="MuiBox-root css-1bvc4cc"
              >
                <button
                  class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeLarge MuiButton-textSizeLarge MuiButtonBase-root  css-zcpmcl-MuiButtonBase-root-MuiButton-root"
                  data-testid="resend-invite-button"
                  tabindex="0"
                  type="button"
                >
                  Resend
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
                <button
                  class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeLarge MuiButton-textSizeLarge MuiButtonBase-root  css-zcpmcl-MuiButtonBase-root-MuiButton-root"
                  data-testid="cancel-invite-button"
                  tabindex="0"
                  type="button"
                >
                  Cancel invite
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </td>
          </tr>
          <tr
            class="MuiTableRow-root css-555rmi-MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-11mkwu5-MuiTableCell-root"
            >
              email2@org.com
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-11mkwu5-MuiTableCell-root"
            >
              10 April 2022
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-11mkwu5-MuiTableCell-root"
            >
              <div
                class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorSuccess MuiChip-filledSuccess css-10gd6cu-MuiChip-root"
              >
                <span
                  class="MuiChip-label MuiChip-labelSmall css-t3ycia-MuiChip-label"
                >
                  Organisation admin
                </span>
              </div>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-11mkwu5-MuiTableCell-root"
            >
              <div
                class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorError MuiChip-filledError css-o7yv1p-MuiChip-root"
              >
                <span
                  class="MuiChip-label MuiChip-labelSmall css-t3ycia-MuiChip-label"
                >
                  Declined
                </span>
              </div>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-11mkwu5-MuiTableCell-root"
            />
          </tr>
        </tbody>
      </table>
      <div
        class="MuiTablePagination-root css-jtlhu6-MuiTablePagination-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar css-17xoeoc-MuiToolbar-root-MuiTablePagination-toolbar"
        >
          <div
            class="MuiTablePagination-spacer css-1psng7p-MuiTablePagination-spacer"
          />
          <p
            class="MuiTablePagination-selectLabel css-1fki1wl-MuiTablePagination-selectLabel"
            id="mui-2"
          >
            Rows per page:
          </p>
          <div
            class="MuiInputBase-root MuiInputBase-colorPrimary css-i13sg4-MuiInputBase-root-MuiTablePagination-select"
            variant="standard"
          >
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-labelledby="mui-2 mui-1"
              class="MuiTablePagination-select MuiSelect-select MuiSelect-standard MuiInputBase-input css-194a1fa-MuiSelect-select-MuiInputBase-input"
              id="mui-1"
              role="button"
              tabindex="0"
            >
              12
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
              tabindex="-1"
              value="12"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTablePagination-selectIcon MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
          <p
            class="MuiTablePagination-displayedRows css-lwa6hs-MuiTablePagination-displayedRows"
          >
            1â€“2 of 2
          </p>
          <div
            class="MuiTablePagination-actions"
          >
            <button
              aria-label="Go to previous page"
              class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-1hk5nzv-MuiButtonBase-root-MuiIconButton-root"
              disabled=""
              tabindex="-1"
              title="Go to previous page"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="KeyboardArrowLeftIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                />
              </svg>
            </button>
            <button
              aria-label="Go to next page"
              class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-1hk5nzv-MuiButtonBase-root-MuiIconButton-root"
              disabled=""
              tabindex="-1"
              title="Go to next page"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="KeyboardArrowRightIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "context": Object {
    "auth": Object {
      "activeRole": "user",
      "allowedRoles": Set {
        "user",
      },
      "changeRole": [MockFunction],
      "defaultRole": "user",
      "getJWT": [MockFunction],
      "loadProfile": [MockFunction],
      "loading": false,
      "login": [MockFunction],
      "logout": [MockFunction],
      "profile": Object {
        "addresses": Array [],
        "attributes": Array [],
        "avatar": "",
        "contactDetails": Array [],
        "createdAt": "",
        "dietaryRestrictions": null,
        "disabilities": null,
        "dob": "2002-04-06",
        "email": "john.smith@example.com",
        "familyName": "Smith",
        "fullName": "John Smith",
        "givenName": "John",
        "id": "cacb559d-b85d-5e64-b623-37252520ebda",
        "jobTitle": "Developer",
        "lastActivity": 2020-01-01T00:00:00.000Z,
        "organizations": Array [],
        "phone": "+44 1234555",
        "preferences": Array [],
        "roles": Array [
          Object {
            "role": Object {
              "name": "user",
            },
          },
        ],
        "tags": Array [],
        "title": "Mr",
        "updatedAt": "",
      },
      "reloadCurrentProfile": [MockFunction],
    },
  },
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
